### PR TITLE
 methods for listing PRs where the commit is head & listing branches which contain the commit

### PR DESCRIPTION
**Pull request by tginiotis-at-work**
# Description 

Methods for listing PRs where the commit is head & listing branches which contain the commit.

# Before submitting a PR:
We love getting PRs, but we hate asking people for the same basic changes every time. 

- [x] Push your changes to a branch other than `master`. Create your PR from that branch.    
- [x] Add JavaDocs and other comments
- [x] Write tests that run and pass in CI. See [CONTRIBUTING.md](CONTRIBUTING.md) for details on how to capture snapshot data.
- [x] Run `mvn clean compile` locally. This may reformat your code, commit those changes.
- [x] Run `mvn -D enable-ci clean install site` locally. If this command doesn't succeed, your change will not pass CI.

# When creating a PR: 

- [x] Fill in the "Description" above. 
- [x] Enable "Allow edits from maintainers".

### CODE REVIEW

**Comment by bitwiseman** [comment]
@tginiotis-at-work 
This looks good so far.

**Comment by tginiotis-at-work** [comment]
@bitwiseman is the PR alright now?

File: src/test/java/org/kohsuke/github/CommitTest.java [hunk 1]
```diff
@@ -25,6 +25,44 @@ public void listFiles() throws Exception {
...
+    public void listPullRequests() throws Exception {
+        GHRepository repo = gitHub.getRepository("hub4j/github-api");
+
+        int[] prsToTest = { 909, 898, 893 };
+
+        for (Integer prNumber : prsToTest) {
+            GHPullRequest pullRequest = repo.getPullRequest(prNumber);
+
+            GHCommit commit = repo
+                    .getCommit(pullRequest.listCommits().toList().stream().findFirst().map(it -> it.sha).orElse(""));
```
[the remark this answers is not in the document]
Comment (tginiotis-at-work) [reply src/test/java/org/kohsuke/github/CommitTest.java]: Reworked the test.

File: src/test/java/org/kohsuke/github/CommitTest.java [hunk 2]
```diff
@@ -25,6 +25,44 @@ public void listFiles() throws Exception {
...
+
+            assertTrue("Pull request " + pullRequest.number + " not found by searching from commit.",
```
[the remark this answers is not in the document]
Comment (tginiotis-at-work) [reply src/test/java/org/kohsuke/github/CommitTest.java]: Implemented the suggestions.

**Comment by tginiotis-at-work** [comment]
I have added the 4 additional test cases & implemented the suggestions. Is there anything more than needs attention?

**Review by bitwiseman — APPROVED** [review]
Excellent!

**Pull request merged**